### DISCUSSION TITLE
What is MyTypescriptExport?

### DISCUSSION THREAD

**Discussion by hillin** [wasm-bindgen/wasm-bindgen#3495] [Q&A] [answered]
Per [documentation](https://rustwasm.github.io/wasm-bindgen/reference/attributes/on-rust-exports/typescript_custom_section.html):

```rust
#[derive(MyTypescriptExport)]
struct Coords {
    latitude: u32,
    longitude: u32,
}
```

> The proc_derive_macro "MyTypescriptExport" can export its own #[wasm_bindgen(typescript_custom_section)] section, which would then be picked up by wasm-bindgen-cli. This would be equivalent to the contents of the TS_APPEND_CONTENT string in the first example.

It seems it's describing a proc macro that does not exist. Is this just an error in the docs, or the said proc macro is being planned? It would be very handy if we have it.

**Comment by daxpedda** [comment, marked as answer]
This is an example on what you could do with `#[wasm_bindgen(typescript_custom_section)]`, in this example a custom derive-macro uses this functionality.

`#[wasm_bindgen(typescript_custom_section)]` exists, `#[derive(MyTypescriptExport)]` doesn't.
